include/libgit4cpp/GitRepository.h

```diff
@@ -32,7 +32,7 @@
 #include <git2.h>
 #include <gul14/escape.h>
 
-#include "libgit4cpp/LibGitPointer.h"
+#include "libgit4cpp/LibGitObject.h"
```

Comment:
I am not entirely sure I follow your arguments for the renaming. In essence, a `LibGitPointer` is a hand-rolled `unique_ptr` for Git objects. Its semantics seem more pointer-like than object-like to me: There is a test for null, and no member functions of the Git object can be called via `.`.

This is just an observation, I can live with both names.

Reply:
Lol, yes. Of course the unique-ptr is generic and thus there can be no better name.
This is not generic at all but holds objects we get from specifically libgit2.

And additionally it will probably get more information, like the failure message.
It does not need to `throw` like in the `feature/remove-debug-output` branch, but we can transport it within and put the work to throw appropriate objects on the user's shoulders.
At least I hoped we we add the throwing thing, but if not we can at least add the non-throwing messages.

> Its semantics seem more pointer-like than object-like to me: There is a test for null, and no member functions of the Git object can be called via

I can not see any test for null. And the whole api is (at least intended to be) alike expected or optional.
Whatever.

The main point is to have aliases for the things, to make the code readable for users.
I always want to have it easy for the users and do not care how it is handled internally, it should show nicely to the outside and be somehow logical for people who never had and never will be looking what is inside.

Assume you are new to some accounting system. You can query all the past transactions. How would you expect the API to look like? Get a `we_have_obscure_pointers<of_more_obscure_types>` or `Transaction` ?

Reply:
Don't get me wrong, I like the aliases. Still, `LibGitPtr` is just a `unique_ptr` to me, and therefore I find that name more logical than `LibGitObject`. But as I wrote, I can live with both.